--- src/net.c
+++ src/net.c
@@ -21,8 +21,10 @@
 #ifndef WIN32
 #  include <arpa/inet.h>
 #  ifndef _AIX
+#  ifdef HAVE_IFADDRS_H
 #    include <ifaddrs.h>
 #  endif
+#  endif
 #  include <netdb.h>
 #  include <netinet/tcp.h>
 #  include <strings.h>
@@ -719,6 +721,7 @@
 
 
 #if !defined(WIN32) && !defined(_AIX)
+#ifdef HAVE_IFADDRS_H
 
 
 static int net__bind_interface(struct mosquitto__listener *listener, struct addrinfo *rp)
@@ -797,6 +800,7 @@
 	}
 }
 #endif
+#endif
 
 
 static int net__socket_listen_tcp(struct mosquitto__listener *listener)
